src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/HelpersWindows.cs

```diff
@@ -327,6 +327,12 @@ public static AlgorithmIdentifier ToAlgorithmIdentifier(this CRYPT_ALGORITHM_IDE
             }
 
             AlgorithmIdentifier algorithmIdentifier = new AlgorithmIdentifier(Oid.FromOidValue(oidValue, OidGroup.All), keyLength);
+            byte[] parameters = new byte[cryptAlgorithmIdentifer.Parameters.cbData];
```

Comment:
We need the parameters from the Win32 PAL so we can decode them. It seemed reasonable to just always set the parameters on the `AlgorithmIdentifier` all the time if they are present. We could take a more conservative approach of only setting the `Parameters` property if the OID is "1.2.840.113549.1.1.7" so we are are copying less data.

Reply:
A switch statement to list all the cases where it says no copy is needed seems fine (RSAES-PKCS1-v1_5... any others?)

Also, this code could be simplified to `parameters = cryptAlgorithmIdentifier.Parameters.ToByteArray()` (extension method earlier in this file).